app/src/main/java/com/mk/tasky/agenda/data/AgendaRepositoryImpl.kt

```diff
@@ -70,22 +70,26 @@ class AgendaRepositoryImpl(
     override fun getAgenda(date: LocalDateTime, forceRemote: Boolean): Flow<Agenda> {
         // TODO: Update with Tasks and Events
         return flow {
-            val localReminders = getRemindersForDate(date.toLocalDate()).toMutableList()
-            emit(Agenda(reminders = localReminders, tasks = emptyList(), events = emptyList()))
+            val localReminders = getRemindersForDate(date.toLocalDate())
+            val localTasks = getTasksForDate(date.toLocalDate())
+            emit(Agenda(reminders = localReminders, tasks = localTasks, events = emptyList()))
             if (forceRemote) {
                 getAgendaRemotely(date).onSuccess { response ->
                     supervisorScope {
                         response.reminders.map {
                             launch { dao.insertReminder(it.toDomain().toEntity()) }
                         }.forEach { it.join() }
+                        response.tasks.map {
+                            launch { dao.insertTask(it.toDomain().toEntity()) }
+                        }.forEach { it.join() }
```

Comment:
Ideally, this should use a separate supervisorScope, else inserting the tasks will wait until all reminders are inserted